### PR TITLE
Set up a "Hello world" project as one app with Webpack

### PR DESCRIPTION
**Pull request by yusufyusie**
## Project requirements

- Create a new Rails app called 'hello-rails-react'.
- Use the command rails new -j webpack my_new_rails_app as described in the previous lesson's tutorial.
- Verify that the jsbundling-rails gem is added to your Gemfile.
- Initialize your project with Git.
- Install needed node dependencies as described in the previous lesson's tutorial.
- Create a babel.config.js in the root folder of your project as described in the previous lesson's tutorial.
- Update webpack.config.js to include the babel loader as described in the previous lesson's tutorial.
- Make sure that your project has a Postgres database set up.
- Create a table for storing your messages and create 5 different greetings.
- Create an API endpoint that selects a random greeting from your table (you will need a controller with an action and Rails route).
- Create a static view that will be the root of your app.

- Create the App component with react-router and render it in your static view.
- Create the Greeting component that will display a greeting. Set it up as a route in your App component.
- Create a store, an action, and a reducer that will connect to your API endpoint to get the random greeting.
- Display the random greeting in your Greeting component.

### CODE REVIEW

**Comment by shahadat3669** [comment]
@yusufyusie, Please do not push new changes after requesting a review. It becomes confusing if you push multiple changes within a short interval. Please refrain from doing so until you get feedback.

File: README.md [hunk 1]
```diff
@@ -1,24 +1,170 @@
...
+
+```sh
+npm install
+```
+
+Build project with:
+
+```sh
+npm run build
+```
```
Comment (shahadat3669) [inline README.md:99]: - [ ] Well! good job so far. but still need some changes. Currenty you added this command to build the project but when I run `npm run build` it give a error. I notic that in the package.json there want no such script. So I will reocmd you to fix this error and make sure your project work properly without any error.

image

**Review by shahadat3669 — CHANGES_REQUESTED** [review]
Hi @yusufyusie, 

Great job so far! 👏 You've done well, and you're almost there for the next project ⏭️, but there are a few issues 🐛 that you still need to work on.




## Don't Worry 🙏

It's okay to have bugs in your code. 🆗 Please see this code review as an opportunity to learn practical coding skills and discover new tips that can improve your code quality 💯 and help you **get even better** 💼.

**I will guide you on how to address these areas down below 👇.**

## Highlights
✅ No linter Errors
✅ Descriptive PR Title
✅ Descriptive PR Description



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

## Have any doubt ❓

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

<table width="100%" align="center">
  <tr>
    <td colspan="4" align="center">
      <strong>Contact me: Shahadat Hossain</strong>
    </td>
  </tr>
  <tr>
    <td align="center">
      <a href="https://github.com/shahadat3669">
        GitHub
      </a>
    </td>
    <td align="center">
      <a href="https://www.linkedin.com/in/shahadat-cseng/">
        LinkedIn
      </a>
    </td>
    <td align="center">
      <a href="https://app.slack.com/client/T47CT8XPG/D05GT4E4WAY/rimeto_profile/U04426SD71V">
        Slack
      </a>
    </td>
    <td align="center">
      <a href="mailto:shahadat3669@gmail.com?subject=Microverse%20Code%20Review">
        Email
      </a>
    </td>
  </tr>
</table>

## ⚠️ WARNING ⚠️

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by shahadat3669** [comment]
@yusufyusie, I have been repeatedly rebuilding the app and providing feedback commits. However, each time I try to finish the review, you push new changes. Then I have to rebuild the entire app and check all the changes again. This has happened three times so far and it is quite confusing. So I have provided my feedback based on the review commit number [29fa1bf] (https://github.com/yusufyusie/hello-rails-react/pull/1/commits/29fa1bfbb8346c55f19b88c7ca49f17e4ab8dfe9) which was before the review was received.

It seems that each time I'm on the verge of completing the review, you push new changes, necessitating a rebuild of the entire app. While I understand the iterative nature of development, this has occurred three times already, leading to some confusion on my end.

To streamline the review process and ensure clarity, I suggest before requesting a review check the changes carefully and stop committing until get feedback from the reviewer.  This would allow both of us to work more efficiently and minimize the need for repeated rebuilds and rechecking everything.

For reference, my feedback is based on the review commit number [29fa1bf] (https://github.com/yusufyusie/hello-rails-react/pull/1/commits/29fa1bfbb8346c55f19b88c7ca49f17e4ab8dfe9), which was the state of the app before the review was initially received.

**Review by Ibizugbe — APPROVED** [review]
Hi @yusufyusie 👋🏼👋🏼👋🏼, 

approved


Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights✅✅✅

- [x] Build script works well👍🏼.
- [x] Linters are passing👍🏼.
- [x] App works fine👍🏼.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**